src/components/pages/home.jsx

```diff
@@ -32,28 +34,32 @@ const HomePage = React.createClass({
     return {inputValue: ""}
   },
   handleClearSearchClicked() {
-    this.setState({inputValue: ""})
-    this.context.router.push("")
+    const query = ""
+    this.setState({inputValue: query})
+    this.context.setQuery(query)
+    this.context.router.push(query)
   },
   handleInputChange(event) {
     this.setState({inputValue: event.target.value})
     this.searchInput.scrollIntoView()
   },
   handleSubmit(event) {
     event.preventDefault()
+    this.context.setQuery(this.state.inputValue)
     this.context.router.push(`?q=${this.state.inputValue}#${searchInputId}`)
```

Comment:
Why build a query string when you can use a much more readable [location descriptor](https://github.com/ReactTraining/react-router/blob/master/docs/Glossary.md#locationdescriptor)?